### ISSUE TITLE
`padding-line-between-statements`: classes and their methods are not `block-like`

### ISSUE THREAD

**Issue by haines** [eslint-stylistic/eslint-stylistic#446]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I would expect classes and their methods to be `block-like` just like function declarations. They aren't.

Given 

```ts
export class Foo {
  public bar(): number {
    return 42;
  }
  public get baz(): string {
    return 'hello';
  }
  public qux = 9000;
}
export function foo(): Foo {
  return new Foo();
}
export function bar(): number {
  return foo().bar();
}
```

with 

```ts
'@stylistic/padding-line-between-statements': [
  'error',
  {
    blankLine: 'always',
    prev: ['block-like'],
    next: '*',
  },
],
```

I would expect blank lines after every `}`, but only one is reported as missing (before `export function bar(): number {`).

Classes can be targeted with the `class` statement type but there doesn't seem to be an equivalent that works for method definitions?

### Reproduction

https://stackblitz.com/edit/github-ua4adn?file=src%2Findex.ts

### Contributes

- [X] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests

**Comment by clemp6r** [comment]
Did you check the `lines-between-class-members` rule?

https://eslint.style/rules/js/lines-between-class-members

**Comment by haines** [comment]
Oh nice, yes, that can achieve what I want.
